distrib/setup.sh

```diff
@@ -31,8 +31,31 @@ get_installation_directory() {
 }
 
 get_from_url() {
-    local url=$1
-    local response=$(curl -k -s -w "%{http_code}" "$url")
+    local server=$1
+    local url=$2
+    if [[ -n "${https_proxy}" ]] || [[ -n "${http_proxy}" ]]; then
+      if [ -n "${https_proxy}" ]; then
+        local proxy=$https_proxy
+      else
+        local proxy=$http_proxy
+      fi
+      local pem_file="${server}.pem"
+      if [ -n "${SL_INSECURE}" ]; then
+        openssl s_client -proxy "$proxy" -showcerts -servername "$server" -connect "${server}:443" </dev/null 2>/dev/null | openssl x509 -outform PEM > "$pem_file" 2>/dev/null
+      else
```

Comment:
Why do we need to use SSL and CA Path if the user did not request an insecure path ?

Reply:
I see that between line 43 and line 51 openssl appears in all paths. Which means that openssl is always invoked even if not necessary

Reply:
Well it didn't work for me at all without